src/V3SchedTiming.cpp

```diff
@@ -115,22 +121,17 @@ AstCCall* TimingKit::createCommit(AstNetlist* const netlistp) {
                 m_commitFuncp->declPrivate(true);
                 scopeTopp->addBlocksp(m_commitFuncp);
             }
+            // Create the commit call and put it in the commit function
             AstSenTree* const sentreep = activep->sentreep();
+            AstIf* const ifp = V3Sched::createIfFromSenTree(sentreep);
+            m_commitFuncp->addStmtsp(ifp);
             FileLine* const flp = sentreep->fileline();
-            // Negate the sensitivity. We will commit only if the event wasn't triggered on the
-            // current iteration
-            auto* const negSentreep = sentreep->cloneTree(false);
-            negSentreep->sensesp()->sensp(
-                new AstLogNot{flp, negSentreep->sensesp()->sensp()->unlinkFrBack()});
-            sentreep->addNextHere(negSentreep);
-            auto* const newactp = new AstActive{flp, "", negSentreep};
-            // Create the commit call and put it in the commit function
-            auto* const commitp = new AstCMethodHard{
-                flp, new AstVarRef{flp, schedulerp, VAccess::READWRITE}, VCMethod::SCHED_COMMIT};
+            AstVarRef* const refp = new AstVarRef{flp, schedulerp, VAccess::READWRITE};
+            AstCMethodHard* const commitp = new AstCMethodHard{flp, refp, VCMethod::SCHED_COMMIT};
             if (resumep->pinsp()) commitp->addPinsp(resumep->pinsp()->cloneTree(false));
             commitp->dtypeSetVoid();
-            newactp->addStmtsp(commitp->makeStmt());
-            m_commitFuncp->addStmtsp(newactp);
```

Comment:
@kbieganski This patch has supposedly equivalent behaviour to before, however in this code, we used to negate only the first SenItem of `negSenTreep`, not the whole SenTree, which might have multiple SenItems. (There are some diffs in the generated output e.g. in `t_timing_events`.) This does seem like a bug in the old version. Is it? If so it would be nice to have a test that fails without this patch (currently everything passes).